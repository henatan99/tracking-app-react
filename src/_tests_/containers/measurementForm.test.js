import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import MeasurementForm from '../../containers/measurementForm';

const mockStore = configureStore([]);

describe('MeasurementForm', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      myState: 'sample text',
    });

    component = renderer.create(
      <Provider store={store}>
        <MeasurementForm />
      </Provider>,
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
