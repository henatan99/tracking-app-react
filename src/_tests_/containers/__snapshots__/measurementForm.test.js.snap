// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeasurementForm should render with given state from Redux store 1`] = `
<div
  className="measurement"
>
  <header
    className="measurement-header"
  >
    <h3>
      , Measure your 
    </h3>
  </header>
  <div
    className="measurement-form-wrapper"
  >
    <form
      className="measurement-form"
      onSubmit={[Function]}
    >
      <input
        className="measurement-input"
        onChange={[Function]}
        placeholder="0.00"
        type="text"
      />
      <button
        className="measurement-button"
        type="submit"
      >
        <span
          className="iconify"
          data-icon="bx:bxs-badge-check"
          data-inline="false"
        />
      </button>
    </form>
    <div
      className="measurement-selector"
    >
      <button
        className="measurement-selector-btn1"
        onClick={[Function]}
        type="button"
      >
        Previous
      </button>
      <button
        className="measurement-selector-btn2"
        onClick={[Function]}
        type="button"
      >
        Next
      </button>
    </div>
  </div>
  <p>
    1
  </p>
</div>
`;
